### PR TITLE
Fixes #3217 #3204 : Added a common interface for dialog popup and share on social media

### LINKED ISSUE #3217: share a common interface for dialog popup and share on social media
**Actual Behaviour**

interface for dialog popup and share on social media differs greatly

**Screenshots of the issue**

Share on social media:

Screenshot 2020-01-05 at 10 16 34 PM

Dialog:

Screenshot 2020-01-05 at 10 15 47 PM

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by anishagg17**
Fixes #3217  #3204

Changes: 
Removed redundant code 
Added a common interface for dialog popup and share on social media
implemented the statefulness of the shareText

Demo Link: https://pr-3220-fossasia-susi-web-chat.surge.sh

Screenshots of the change: 
Screenshot 2020-01-05 at 10 15 00 PM

### CODE REVIEW

**Comment by anishagg17** [comment]
Screenshot 2020-01-07 at 2 32 16 PM

Updated width

**Comment by anishagg17** [comment]
`Also, can you implement the statefulness of the shareText in this PR itself.`

Done Sir!

File: src/components/Settings/ShareOnSocialMedia.js [hunk 1]
```diff
@@ -55,56 +56,82 @@ const TelegramShareButton = styled(_TelegramShareButton)`
...
 `;
 
-const ShareOnSocialMedia = () => {
-  const shareUrl = 'https://susi.ai';
-  const title = 'Lets chat with SUSI.AI, the open source personal assistant';
-  const shareText =
-    'Lets chat with SUSI.AI, the open source personal assistant https://susi.ai';
+class ShareOnSocialMedia extends Component {
+  state = {
+    shareUrl: 'https://susi.ai',
```
[the remark this answers is not in the document]
Comment (anishagg17) [reply src/components/Settings/ShareOnSocialMedia.js]: `You need to add a onChange function when the user changes the text in the input field.
`
Done Sir!

File: src/components/Settings/ShareOnSocialMedia.js [hunk 2]
```diff
@@ -55,56 +56,88 @@ const TelegramShareButton = styled(_TelegramShareButton)`
...
-            </div>
-          </TelegramShareButton>
-        </IconButton>
-      </ShareIconContainer>
-    </React.Fragment>
-  );
+  render() {
+    return (
+      <React.Fragment>
+        <DialogTitle>Share about SUSI</DialogTitle>
```
Comment (akshatnitd) [inline src/components/Settings/ShareOnSocialMedia.js]: Make it `SUSi.AI`

File: src/components/Settings/ShareOnSocialMedia.js [hunk 3]
```diff
@@ -55,56 +56,88 @@ const TelegramShareButton = styled(_TelegramShareButton)`
...
 `;
 
-const ShareOnSocialMedia = () => {
-  const shareUrl = 'https://susi.ai';
-  const title = 'Lets chat with SUSI.AI, the open source personal assistant';
-  const shareText =
-    'Lets chat with SUSI.AI, the open source personal assistant https://susi.ai';
+class ShareOnSocialMedia extends Component {
+  state = {
+    shareUrl: 'https://susi.ai',
```
Comment (akshatnitd) [inline src/components/Settings/ShareOnSocialMedia.js]: Store `https://susi.ai` in a constant and use here.

File: src/components/Settings/ShareOnSocialMedia.js [hunk 4]
```diff
@@ -55,56 +56,88 @@ const TelegramShareButton = styled(_TelegramShareButton)`
...
+    title:
+      this.props.title ||
+      'Lets chat with SUSI.AI, the open source personal assistant',
+    shareText:
+      'Lets chat with SUSI.AI, the open source personal assistant https://susi.ai',
```
Comment (akshatnitd) [inline src/components/Settings/ShareOnSocialMedia.js]: Reuse constant here as well.

Comment (anishagg17) [reply src/components/Settings/ShareOnSocialMedia.js]: Done Sir!

File: src/components/Settings/ShareOnSocialMedia.js [hunk 5]
```diff
@@ -55,56 +56,91 @@ const TelegramShareButton = styled(_TelegramShareButton)`
...
-        </IconButton>
-      </ShareIconContainer>
-    </React.Fragment>
-  );
+  handleChange = event => this.setState({ shareText: event.target.value });
+
+  render() {
+    return (
+      <React.Fragment>
+        <DialogTitle>Share about SUSi.AI</DialogTitle>
```
Comment (akshatnitd) [inline src/components/Settings/ShareOnSocialMedia.js]: It should be `SUSI.AI`. That was a typo in the previous comment. 

Also, you must think logically before following any feedback.

File: src/components/Settings/ShareOnSocialMedia.js [hunk 6]
```diff
@@ -55,56 +56,91 @@ const TelegramShareButton = styled(_TelegramShareButton)`
...
 
-  return (
-    <React.Fragment>
-      <DialogTitle>Share about SUSI</DialogTitle>
+class ShareOnSocialMedia extends Component {
+  state = {
+    shareUrl,
+    title:
+      this.props.title ||
+      'Lets chat with SUSI.AI, the open source personal assistant',
```
Comment (akshatnitd) [inline src/components/Settings/ShareOnSocialMedia.js]: shareText and the title string in common. Please reuse them by shareText + url.

File: src/components/Settings/ShareOnSocialMedia.js [hunk 7]
```diff
@@ -55,56 +56,91 @@ const TelegramShareButton = styled(_TelegramShareButton)`
   ${commonIconStyle};
 `;
 
-const ShareOnSocialMedia = () => {
-  const shareUrl = 'https://susi.ai';
-  const title = 'Lets chat with SUSI.AI, the open source personal assistant';
-  const shareText =
-    'Lets chat with SUSI.AI, the open source personal assistant https://susi.ai';
+const shareUrl = 'https://susi.ai';
```
Comment (akshatnitd) [inline src/components/Settings/ShareOnSocialMedia.js]: This URL must be present in constants folder. Use it from there.

File: src/components/Settings/ShareOnSocialMedia.js (hunk 6 above)
Comment (anishagg17) [reply src/components/Settings/ShareOnSocialMedia.js]: Done sir!

File: src/components/Settings/ShareOnSocialMedia.js [hunk 8]
```diff
@@ -55,56 +57,88 @@ const TelegramShareButton = styled(_TelegramShareButton)`
...
+  'Lets chat with SUSI.AI, the open source personal assistant';
 
-  return (
-    <React.Fragment>
-      <DialogTitle>Share about SUSI</DialogTitle>
+class ShareOnSocialMedia extends Component {
+  state = {
+    shareUrl,
+    title: this.props.title || defaultShareTitle,
+    shareText: defaultShareTitle + shareUrl,
```
Comment (akshatnitd) [inline src/components/Settings/ShareOnSocialMedia.js]: Add space between text and url.

File: src/constants/about.js [hunk 9]
```diff
@@ -104,3 +104,5 @@ export const DEVICES_DATA = [
     videoSrc: susiDesktop,
   },
 ];
+
+export const shareUrl = 'https://susi.ai';
```
Comment (akshatnitd) [inline src/constants/about.js]: Do not make a new variable. Use variable from https://github.com/fossasia/susi.ai/blob/b0b666be89192b478dc69875607421cb67870279/src/utils/urls.js#L13

Comment (anishagg17) [reply src/constants/about.js]: Done Sir!

**Pull request merged**